include/bluetooth/controller.h

```diff
@@ -29,6 +29,16 @@ extern "C" {
  */
 void bt_ctlr_set_public_addr(const u8_t *addr);
 
+/** @brief Set dynamically Bluetooth Tx power
+ * 
+ *  Should be called after bt_enable() for chances to take effect.
+ * 
+ *  @param power Tx power level desired. 
+ * 
+ *  @return Zero on success or EINVAL error code otherwise.
+ */
+int bt_ctrl_set_tx_power(const u32_t power);
```

Comment:
As already mentioned by others, a vendor HCI makes more sense for tx power control.